common/set.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import "encoding/json"
+
+// Set enables deduplication of objects parsed by JQArrayParser on Add()
+type Set[T any] struct {
+	data map[string]T
+}
+
+func NewSet[T any]() *Set[T] {
+	return &Set[T]{
+		data: make(map[string]T),
+	}
+}
+
+// Add performs deduplication of objects in the Set by treating the json marshaled value as a UID
```

Comment:
Removed uniquemap.go because I realized a set is what we need.

The problem with using a uniquemap.go is that it is unreliable - the JSON marshal of the object you're saving is the thing that guarantees uniqueness. 

Using a unique ID other than the JSON marshal of the object is prone to errors.